### PR TITLE
build(docs-infra): ensure that deployment works on CI

### PR DESCRIPTION
**Pull request by petebacondarwin**
The actual "main" part of the script that is executed was using
an uninitialized variable. This is fixed and a test is added to
check.

### CODE REVIEW

File: aio/scripts/deploy-to-firebase.spec.js [hunk 1]
```diff
@@ -259,4 +260,24 @@ describe('deploy-to-firebase:', () => {
...
+  it('integration - should run the main script without error', () => {
+    const cmd = `"${process.execPath}" "${__dirname}/deploy-to-firebase" --dry-run`;
+    const env = {
+      CI_REPO_OWNER: 'angular',
+      CI_REPO_NAME: 'angular',
+      CI_PULL_REQUEST: 'false',
+      CI_BRANCH: 'master',
+      CI_COMMIT: getLatestCommit('master')
+    };
+    const result = execSync(cmd, {encoding: 'utf8', env}).toString().trim();
```
Comment (gkalpak) [inline aio/scripts/deploy-to-firebase.spec.js:273]: Nit: You don't need both `encoding: 'utf8'` and `.toString()`. One of the two is enough.
<sub>
(Without `encoding: 'utf8'` `execSync()` returns a `Buffer`, so you need `.toString()` to convert it to a string. With `encoding: 'utf8'` `execSync()` will return a `String`, so no need for `.toString()`.)
</sub>

**Review by gkalpak — APPROVED** [review]
Thx for cleaning up my mess 😇

File: aio/scripts/deploy-to-firebase.spec.js (hunk 1 above)
Comment (petebacondarwin) [reply aio/scripts/deploy-to-firebase.spec.js]: Gotcha!

**Comment by AndrewKushnir** [comment]
@gkalpak should we backport this fix to the 9.1.x branch as well?

**Comment by petebacondarwin** [comment]
> @gkalpak should we backport this fix to the [9.1.x](https://github.com/angular/angular/commits/9.1.x) branch as well?

Yes and 8.2.x :-) I was waiting for approval before creating the backports

**Comment by AndrewKushnir** [comment]
Thanks @petebacondarwin 👍

**Pull request closed without merging**